Validation/RecoTrack/plugins/MultiTrackValidator.cc

```diff
@@ -459,6 +461,44 @@ size_t MultiTrackValidator::tpDR(const TrackingParticleRefVector& tPCeff,
   return n_selTP_dr;
 }
 
+
+size_t MultiTrackValidator::tpDR_jet(const TrackingParticleRefVector& tPCeff,
+                                 const std::vector<size_t>& selected_tPCeff,
+                                 DynArray<float>& dR_tPCeff_jet,
+                                 const edm::View<reco::Candidate>* const & cores) const {
```

Comment:
Maybe this function could be merged with the `tpDR` above by passing the `dR_tPCeff_jet` and `cores` by pointer, and fill the former only if the latter is `nullptr`? The logic of the loops is rather heavy to be copy-pasted.

Reply:
I thought about that but the only copy-pasted part is the loop over selected_tPCeff i.e. the following lines:

`float etaL[tPCeff.size()], phiL[tPCeff.size()];
  size_t n_selTP_dr = 0;
  for(size_t iTP: selected_tPCeff) {
    //calculare dR wrt inclusive collection (also with PU, low pT, displaced)
    auto const& tp2 = *(tPCeff[iTP]);
    auto  && p = tp2.momentum();
    etaL[iTP] = etaFromXYZ(p.x(),p.y(),p.z());
    phiL[iTP] = atan2f(p.y(),p.x());
  }
  for(size_t iTP1: selected_tPCeff) {
    auto const& tp = *(tPCeff[iTP1]);
    double dR = std::numeric_limits<double>::max();
    if(dRtpSelector(tp)) { `

Instead between the two functions change the return (void VS size_t), arguments (in  tpDR_jet there is also the cores), so it is not easy to merge. If you think it is mandatory I can:
1) split the original tpDR in two functions: the first one does the first loop, the second one does the second loop
2)  tpDR_jet start from the first-loop-function and apply a second different loop. 
In the end, we will have three functions: one common loop and two for the different part. 

Anyway, in my opinion, is more clear in the old current way.

Reply:
ok, you are right. I implemented the function as you suggested